lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1692,8 +1692,8 @@ void AsmPrinter::EmitPGLT(const GlobalVariable *GV) {
   MCSymbol *GOTSymbol = OutContext.getOrCreateSymbol(StringRef("_GLOBAL_OFFSET_TABLE_"));
   OutStreamer->EmitValue(MCSymbolRefExpr::create(GOTSymbol, OutContext), PtrSize);
 
-  for (const MCSymbol *Sym : OutContext.getBinSymbols())
-    OutStreamer->EmitValue(MCSymbolRefExpr::create(Sym, OutContext), PtrSize);
+  for (auto *Sec : PGLT)
+    OutStreamer->EmitValue(MCSymbolRefExpr::create(Sec->getBeginSymbol(), OutContext), PtrSize);
```

Comment:
Would this still work with a default bin/section? If we have a default bin/section #0 could we treat functions uniformly and avoid `F->isRandPage()` guards?!